src/Lecture3.hs

```diff
@@ -159,11 +177,14 @@ monsters, you should get a combined treasure and not just the first
 🕯 HINT: You may need to add additional constraints to this instance
   declaration.
 -}
-instance Semigroup (Treasure a) where
-
+instance Semigroup a => Semigroup (Treasure a) where
+    b              <> NoTreasure     = b
+    NoTreasure     <> a              = a
+    SomeTreasure a <> SomeTreasure b = SomeTreasure (a <> b)
 
-instance Monoid (Treasure a) where
 
+instance Monoid a => Monoid (Treasure a) where
```

Comment:
The `Monoid` constraint is too strict here. You don't use `mempty` so `Semigroup` should be enough 👌🏻 

```suggestion
instance Semigroup a => Monoid (Treasure a) where
```

Reply:
Hm 🤔